devtools/test_dashboard_v2/devtools.js

```diff
@@ -0,0 +1,138 @@
+// Our gracious testing object
+var Tabs = {
+
+    getGraph: function(id) {
+        id = id || 'default-plot';
+        return document.getElementById(id);
+    },
+
+    fresh: function(id) {
+        id = id || 'default-plot';
+
+        var graphDiv = Tabs.getGraph(id);
+
+        if(graphDiv) {
+            graphDiv.remove();
+        }
+
+        graphDiv = document.createElement('div');
+        graphDiv.className = 'dashboard-plot';
+        graphDiv.id = id;
+
+        var plotArea = document.getElementById('plots');
+        plotArea.appendChild(graphDiv);
+
+        return graphDiv;
+    },
+
+    plotMock: function(mockName, id) {
+        var mockURL = '/test/image/mocks/' + mockName + '.json';
+
+        window.Plotly.d3.json(mockURL, function(err, fig) {
+            window.Plotly.plot(Tabs.fresh(id), fig.data, fig.layout);
+        });
+    },
+
+    snapshot: function(id) {
+        var gd = Tabs.getGraph(id);
+
+        if(!gd._fullLayout || !gd._fullData) {
+            return;
+        }
+
+        var image = new Image();
+
+        window.Plotly.Snapshot.toImage(gd, { format: 'png' }).on('success', function(img) {
+            image.src = img;
+
+            var imageDiv = document.getElementById('snapshot');
+            imageDiv.appendChild(image);
+
+            console.warn('Snapshot complete!');
+        });
+    },
+
+    purge: function() {
+        var plots = document.getElementsByClassName('dashboard-plot');
+        var images = document.getElementsById('snapshot');
+
+        while(images.firstChild) {
+            images.removeChild(images.firstChild);
+        }
+
+        for(var i = 0; i < plots.length; i++) {
+            window.Plotly.purge(plots[i]);
+        }
+    },
+
+    onReload: function() {
+        return;
+    },
+
+    reload: function() {
+        var source = document.getElementById('source');
+        var reloaded = document.getElementById('reload-time');
+
+        source.remove();
+
+        window.Plotly = null;
+
+        source = document.createElement('script');
+        source.id = 'source';
+        source.src = '../../build/plotly.js';
+
+        document.body.appendChild(source);
+
+        var reloadTime = new Date().toLocaleTimeString();
+        console.warn('plotly.js reloaded at ' + reloadTime);
+        reloaded.textContent = 'last reload at ' + reloadTime;
+
+        Tabs.onReload();
+    }
+};
+
+
+setInterval(function() {
+    window.gd = Tabs.getGraph() || Tabs.fresh();
+    window.fullLayout = window.gd._fullLayout;
+    window.fullData = window.gd._fullData;
+}, 1000);
```

Comment:
I frequently inspect fullData and fullLayout, so this just keeps them handy on the window object. 

Nice for quick `Plotly.____(gd, ___, ___)` operations and checking values without having to get the object every time.